File: android/src/main/kotlin/dev/eggnstone/chime/views/ChimeDefaultVideoRenderViewFactory.kt
package dev.eggnstone.chime.views

import android.content.Context
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory
import java.util.*

class ChimeDefaultVideoRenderViewFactory : PlatformViewFactory(StandardMessageCodec.INSTANCE)
{
    override fun create(context: Context?, viewId: Int, args: Any?): PlatformView
    {
        val view = ChimeDefaultVideoRenderView(context)
        _viewIdToViewMap[viewId] = view
        return view
    }

    companion object
    {
        private val _viewIdToViewMap: MutableMap<Int, ChimeDefaultVideoRenderView> = HashMap()

        fun getViewById(id: Int): ChimeDefaultVideoRenderView? = _viewIdToViewMap[id]

        fun clearViewIds()
        {
            _viewIdToViewMap.clear()
        }
    }
}

File: ios/Classes/views/ChimeDefaultVideoRenderViewFactory.swift
import UIKit
import Flutter

public class ChimeDefaultVideoRenderViewFactory : NSObject, FlutterPlatformViewFactory {
    
    public static var _viewIdToViewMap = [Int64: ChimeDefaultVideoRenderView]()
    
    public func create (
        withFrame frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?
    ) -> FlutterPlatformView {
        let view = ChimeDefaultVideoRenderView(frame, viewId: viewId, args: args)
        ChimeDefaultVideoRenderViewFactory._viewIdToViewMap[viewId] = view
        return view
    }

    public static func getViewById(id: Int64) -> ChimeDefaultVideoRenderView? {
        return self._viewIdToViewMap[id]!
    }
    
    
}
